cmd/libsnap-confine-private/infofile-test.c

```diff
@@ -160,6 +161,94 @@ static void test_infofile_get_key(void) {
     g_assert_null(tricky_value);
     sc_error_free(err);
     fclose(stream);
+
+    /* Unexpected section */
+    char tricky6[] = "[section]\n";
+    stream = fmemopen(tricky6, sizeof tricky6 - 1, "r");
+    g_assert_nonnull(stream);
+    rc = sc_infofile_get_key(stream, "key", &tricky_value, &err);
+    g_assert_cmpint(rc, ==, -1);
+    g_assert_nonnull(err);
+    g_assert_cmpstr(sc_error_domain(err), ==, SC_LIBSNAP_DOMAIN);
+    g_assert_cmpint(sc_error_code(err), ==, 0);
+    g_assert_cmpstr(sc_error_msg(err), ==, "line 1 contains unexpected section");
+    g_assert_null(tricky_value);
+    sc_error_free(err);
+    fclose(stream);
 }
 
-static void __attribute__((constructor)) init(void) { g_test_add_func("/infofile/get_key", test_infofile_get_key); }
+static void test_infofile_get_ini_key(void) {
+    int rc;
+    sc_error *err;
+
+    char text[] =
+        "[section1]\n"
+        "key=value\n"
+        "[section2]\n"
+        "key2=value-two\n"
```

Comment:
can you add `key=value-one-two` to `[section2]` and test that `sc_infofile_get_ini_section_key(stream, "section2", "key", &value, &err);` returns the correct value?

Reply:
Added